src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -686,12 +688,15 @@ public static int findFirstSlotInProviderNotEmptyWith(@NotNull final ICapability
      */
     public static int findFirstSlotInItemHandlerNotEmptyWith(@NotNull final IItemHandler itemHandler, @NotNull final Predicate<ItemStack> itemStackSelectionPredicate)
     {
-        @NotNull final Predicate<ItemStack> notEmptyPredicate = itemStack -> !InventoryUtils.isItemStackEmpty(itemStack);
-        notEmptyPredicate.and(itemStackSelectionPredicate);
+        @NotNull final Predicate<ItemStack> emptyPredicate = InventoryUtils::isItemStackEmpty;
+        @NotNull final Predicate<ItemStack> notEmptyPredicate = emptyPredicate.negate();
+        //todo: extract the first to as static fields
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)